ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java

```diff
@@ -280,6 +283,66 @@ public void cleanup() throws InterruptedException, IOException, StoreException {
   }
 
   /**
+   * Tests blob store use of {@link ClusterManagerWriteStatusDelegate}
+   * @throws StoreException
+   */
+  @Test
+  public void testClusterManagerWriteStatusDelegateUse() throws StoreException, IOException, InterruptedException {
+    //Setup threshold test properties, replicaId, mock write status delegate
+    properties.setProperty(StoreConfig.storeReadOnlyEnableSizeThresholdPercentageName, "65");
+    properties.setProperty(StoreConfig.storeReadWriteEnableSizeThresholdPercentageDeltaName, "5");
+    StoreConfig config = new StoreConfig(new VerifiableProperties(properties));
+    StoreTestUtils.MockReplicaId replicaId = getMockReplicaId(tempDirStr);
+    ClusterManagerWriteStatusDelegate clusterManagerWriteStatusDelegate = mock(ClusterManagerWriteStatusDelegate.class);
+    when(clusterManagerWriteStatusDelegate.unseal(anyObject())).thenReturn(true);
+    when(clusterManagerWriteStatusDelegate.seal(anyObject())).thenReturn(true);
+
+    //Restart store with new threshold properties, mock delegate
+    store.shutdown();
+    store = createBlobStore(replicaId, config, clusterManagerWriteStatusDelegate);
+    store.start();
+
+    //Check that after start, because StoreDescriptor file already exists, delegate is not called
+    verifyZeroInteractions(clusterManagerWriteStatusDelegate);
+
+    //Verify that after putting in enough data, the store goes to read only
+    List<MockId> addedIds = put(5, 900, Utils.Infinite_Time);
```

Comment:
can we first put a few that don't push it over the threshold and make sure it doesn't go to RO?

Reply:
added